x-pack/plugins/security/public/views/management/roles_grid/components/roles_grid_page.tsx

```diff
@@ -0,0 +1,312 @@
+  EuiTitle,
+} from '@elastic/eui';
+import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n/react';
+import _ from 'lodash';
+import React, { Component } from 'react';
+import { toastNotifications } from 'ui/notify';
+import { Role } from '../../../../../common/model/role';
+import { isRoleEnabled } from '../../../../lib/role';
+import { RolesApi } from '../../../../lib/roles_api';
+import { ConfirmDelete } from './confirm_delete';
+import { PermissionDenied } from './permission_denied';
+
+interface Props {
+  intl: InjectedIntl;
+}
+
+interface State {
+  roles: Role[];
+  selection: Role[];
+  filter: string;
+  showDeleteConfirmation: boolean;
+  permissionDenied: boolean;
+}
+
+const getRoleManagementHref = (roleName?: string) => {
+  return `#/management/security/roles/edit${roleName ? `/${roleName}` : ''}`;
+};
+
+class RolesGridPageUI extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      roles: [],
+      selection: [],
+      filter: '',
+      showDeleteConfirmation: false,
+      permissionDenied: false,
+    };
+  }
+
+  public componentDidMount() {
+    this.loadRoles();
+  }
+
+  public render() {
+    const { permissionDenied } = this.state;
+
+    return permissionDenied ? <PermissionDenied /> : this.getPageContent();
+  }
+
+  private getPageContent = () => {
+    const { roles } = this.state;
+    const { intl } = this.props;
+    return (
+      <EuiPageContent>
+        <EuiPageContentHeader>
+          <EuiPageContentHeaderSection>
+            <EuiTitle>
+              <h2>
+                <FormattedMessage
+                  id="xpack.security.management.roles.roleTitle"
+                  defaultMessage="Roles"
+                />
+              </h2>
+            </EuiTitle>
+            <EuiText color="subdued" size="s">
+              <p>
+                <FormattedMessage
+                  id="xpack.security.management.roles.subtitle"
+                  defaultMessage="Apply roles to groups of users and manage permissions across the stack"
+                />
+              </p>
+            </EuiText>
+          </EuiPageContentHeaderSection>
+          <EuiPageContentHeaderSection>
+            <EuiButton data-test-subj="createRoleButton" href={getRoleManagementHref()}>
+              <FormattedMessage
+                id="xpack.security.management.roles.createRoleButtonLabel"
+                defaultMessage="Create role"
+              />
+            </EuiButton>
+          </EuiPageContentHeaderSection>
+        </EuiPageContentHeader>
+        <EuiPageContentBody>
+          {this.state.showDeleteConfirmation ? (
+            <ConfirmDelete
+              onCancel={this.onCancelDelete}
+              rolesToDelete={this.state.selection.map(role => role.name)}
+              callback={this.handleDelete}
+            />
+          ) : null}
+
+          {
+            // @ts-ignore missing rowProps typedef
+            <EuiInMemoryTable
```

Comment:
Since this table's content is so minimal you can turn off the responsive layout with `responsive = false`